nvm.sh

```diff
@@ -1870,9 +1871,11 @@ nvm_get_arch() {
     *) NVM_ARCH="${HOST_ARCH}" ;;
   esac
 
-  # If running a 64bit ARM kernel but a 32bit ARM userland, change ARCH to 32bit ARM (armv7l)
-  L=$(ls -dl /sbin/init 2>/dev/null) # if /sbin/init is 32bit executable
-  if [ "$(uname)" = "Linux" ] && [ "${NVM_ARCH}" = arm64 ] && [ "$(od -An -t x1 -j 4 -N 1 "${L#*-> }")" = ' 01' ]; then
+  # If running a 64bit ARM kernel but a 32bit ARM userland,
+  # change ARCH to 32bit ARM (armv7l) if /sbin/init is 32bit executable
+  if [ "$(uname)-${NVM_ARCH}" = "Linux-arm64" ] &&
+    L="$(ls -dl /sbin/init 2>/dev/null)" &&
+    [ "$(od -An -t x1 -j 4 -N 1 "${L#*-> }")" = ' 01' ]; then
```

Comment:
i'd rather do this in two checks, just in case one of them has a dash in it.
```suggestion
    if [ "$(uname)" = 'Linux' ] && [ "${NVM_ARCH}" = 'arm64' ] \
      && L="$(ls -dl /sbin/init 2>/dev/null)" \
      && [ "$(od -An -t x1 -j 4 -N 1 "${L#*-> }")" = ' 01' ]; then
```

Reply:
No worries, I'll revert that part of the `if` statement. If either part includes a hyphen, the test fails regardless, even if the other part is empty, but less ambiguous code is almost always a good idea 👍🏻   (I only made the change to shorten the statement, but ended up breaking it onto multiple lines anyway.)